platform/v07/platform.go

```diff
@@ -5,22 +5,22 @@ import (
 	"github.com/buildpacks/lifecycle/platform/common"
 )
 
-type Platform struct {
+type v07Platform struct {
```

Comment:
Why did you rename all of these structures (`pre06Platform`, `v06Platform` and `v07Platform`)?
I think the previous name was better since the name of the package is based on the platform API (`v07` in this case).

In addition, not specifically related to this PR, but I saw in `platform/platform.go` that we create all of the platform even if we don't need them. I don't think it's a good practice although in the current situation it doesn't really matter because each of these structures is pretty small.

Reply:
I found that having the concrete platform have the same name as the interface was confusing. Additionally, there isn't a need for the struct to be public.